[WIP] Add tests

jason-o-matic opened the pull request:
This uses gemika to test multiple versions of ruby and dependencies.

Also add `script/setup` and `script/test` as well as a travis config.

jason-o-matic committed:
Add tests

This uses gemika to test multiple versions of ruby and dependencies.

Also add `script/setup` and `script/test` as well as a travis config.

jason-o-matic committed:
Add mysql to  `script/setup`

Since we test against the mysql gem.

jason-o-matic committed:
Fix issue with `script/setup` exiting prematurely

If the postgres database existed the script would exit, so this handles
that case.

jason-o-matic committed:
Update `script/setup` to check for a database config

jason-o-matic committed:
Testing a travis fix

Gemika recommended the previous config, but it stalled on travis so
I'll try the default config and see if it works.

jason-o-matic committed:
Fix DB names in travis config

Copy paste error

jason-o-matic committed:
Fix travis DB config

Didn't update the copy paste from gemika

jason-o-matic committed:
Add Travis badge

janxious reviewed (approved):
Does what it says on the tin 👍

janxious commented on spec/support/database.rb:
```diff
@@ -0,0 +1,32 @@
+    t.string :city
+  end
+
+  create_table :recipes do |t|
```
probably don't need these

janxious commented on spec/support/models.rb:
```diff
@@ -0,0 +1,35 @@
+end
+
+
+class Recipe < ActiveRecord::Base
```
and probably don't need these

jason-o-matic committed:
Remove unnecessary test models

These were copied and pasted from the gemika setup, but we only really
need one model.

jason-o-matic committed:
Fix rspec deprecation warnings

We want to use the older syntax because it works with older versions of
rspec that we might include in older test Gemfiles. If we used the new
syntax old Gemfiles migth fail because their versions of rspec wouldn't
support the new syntax.

jason-o-matic committed:
Add a test for DelayedJob instrumentation
